tracer/test/Datadog.Trace.Tests/ApiTests.cs

```diff
@@ -128,7 +134,17 @@ public async Task SetsDefaultSamplingRates()
         [InlineData("", false, false)] // Version check fail, partial flush disabled
         public void LogPartialFlushWarning(string agentVersion, bool partialFlushEnabled, bool expectedResult)
         {
-            var api = new Api(new Uri("http://127.0.0.1:1234"), apiRequestFactory: null, statsd: null, updateSampleRates: null, isPartialFlushEnabled: partialFlushEnabled);
+            var responseMock = new Mock<IApiResponse>();
+            responseMock.Setup(x => x.StatusCode).Returns(200);
+
+            var requestMock = new Mock<IApiRequest>();
+            requestMock.Setup(x => x.PostAsync(It.IsAny<ArraySegment<byte>>(), It.IsAny<string>())).ReturnsAsync(responseMock.Object);
+
+            var factoryMock = new Mock<IApiRequestFactory>();
+            factoryMock.Setup(x => x.Create(It.IsAny<Uri>())).Returns(requestMock.Object);
```

Comment:
Is `IApiRequestFactory.Create` invoked? Can we omit all this mocking?

Reply:
> Is IApiRequestFactory.Create invoked
Yes, it is invoked in the constructor of `Api`, so unfortunately we can't avoid it 😞